components/browser/state/src/test/java/mozilla/components/browser/state/store/BrowserStoreExceptionTest.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.store
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.support.test.ext.joinBlocking
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.shadows.ShadowLooper
+import java.lang.IllegalStateException
+
+// These tests are in a separate class because they needs to run with
+// Robolectric (different runner, slower) while all other tests only
+// need a Java VM (fast).
+@RunWith(AndroidJUnit4::class)
+class BrowserStoreExceptionTest {
+
+    @Test(expected = java.lang.IllegalArgumentException::class)
+    fun `AddTabAction - Exception is thrown if parent doesn't exist`() {
+        try {
+            val store = BrowserStore()
+            val parent = createTab("https://www.mozilla.org")
+            val child = createTab("https://www.firefox.com")
+
+            store.dispatch(TabListAction.AddTabAction(child, parent = parent)).joinBlocking()
+
+            // Wait for the main looper to process the re-thrown exception.
+            ShadowLooper.idleMainLooper()
+        } catch (e: IllegalStateException) {
+            val cause = e.cause
+            if (cause != null) {
+                throw cause
+            }
```

Comment:
This part makes me wonder if we should create a `StoreException` just to make sure we are catching the right thing in our tests. Related: I used `RuntimException` before and my tests passed. But it turned out it was a RuntimeException from the tests runner (I wasn't using Robolectric and therefore there was no main looper).

Reply:
OK, let's chat our lib-state to store to throw a `StoreException`. I will amend a commit here with that change.

Reply:
Updated.